File: my-folder/0003-longest-substring-without-repeating-characters/solution.java
class Solution {
    public int lengthOfLongestSubstring(String s) {

        int maxLen=0;
        int l=0;
        Map<Character,Integer> mp=new HashMap<>();

        if(s==null||s.isEmpty()){
            return 0;
        }


        for(int r=0;r<s.length();r++){
            Character c=s.charAt(r);
            if(mp.containsKey(c) && mp.get(c)>=l){
                l=mp.get(c)+1;
                mp.put(c,r);
            }
             mp.put(c,r);           

            // System.out.println(mp);

            maxLen=Math.max(r-l+1,maxLen);
            // System.out.println(maxLen);
        }
        return maxLen;
        
    }
}

File: my-folder/0003-longest-substring-without-repeating-characters/solution.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        i=0
        j=0
        ml=0

        visited={ch:False for ch in s}

        for j in range(len(s)):
            while visited[s[j]]==True:
                #print("Inside sliding")
                visited[s[i]]=False
                i=i+1
                #Do 
            ml=max(j-i+1,ml)
            #print(ml)
            visited[s[j]]=True

        return ml

            


        
